src/classes/CON_ContactMerge_TEST.cls

```diff
@@ -42,306 +42,427 @@ public class CON_ContactMerge_TEST {
+        //Create contacts for going through the contact merge page
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+
+		Contact con2 = UTIL_UnitTestData_TEST.getContact();
+		con2.LastName = '2Contact_forTests_Merge';
+
+        insert new Contact[]{con, con2};
+
+        // Find the ReadOnly profile (a Salesforce standard profile) and create a dummy User with that Profile
+        User readOnlyUser = UTIL_UnitTestData_TEST.createUser(UTIL_UnitTestData_TEST.PROFILE_READONLY_USER);
+
+        // Verify that controller detects the User doesn't have delete permissions on the Contact object
+        Test.startTest();
+        System.runAs(readOnlyUser) {
+            Test.setCurrentPageReference(new PageReference('Page.CON_ContactMerge'));
+            Con_ContactMerge_CTRL ctlr = new Con_ContactMerge_CTRL(new ApexPages.StandardSetController(new list<Contact>()));
+            System.debug('Here is the error msg:'+ ctlr.errorMsg);
+            System.assertEquals(System.Label.conMergeErrorNoDeleteObjPermission, ctlr.errorMsg);
+            UTIL_UnitTestData_TEST.assertPageHasError(System.Label.conMergeErrorNoDeleteObjPermission);
+        }
+        Test.stopTest();
+    }
+
+    /**
+     * @description Validate that a User with delete permissions to the Contact object, but no record level acesss
+       gets a pageErrorMessage doing the contact merge.
+     */
+     @isTest
+     private static void shouldDisplayErrorWithNoDeleteRecAccess() {
+        //Create contacts for going through the contact merge page
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+
+		Contact con2 = UTIL_UnitTestData_TEST.getContact();
+		con2.LastName = '2Contact_forTests_Merge';
+
+        insert new Contact[]{con, con2};
+
+        // Find the ReadOnly profile (a Salesforce standard profile) and create a dummy User with that Profile
+        User readOnlyUser = UTIL_UnitTestData_TEST.createUser(UTIL_UnitTestData_TEST.PROFILE_READONLY_USER);
+
+        User adminUser = UTIL_UnitTestData_TEST.createUserWithoutInsert(UTIL_Profile.SYSTEM_ADMINISTRATOR);
+        // Run as admin user to set the permission set with object level delete permission for the readonlyuser
+        System.runas(adminUser) {
+            assignDeletePermissionSet(readOnlyUser.Id);
+        }
+
+        Test.startTest();
+        // Verify that when logged in as readonlyuser, it returns error when merging contacts because of record level permissions
+        System.runAs(readOnlyUser) {
+            Test.setCurrentPageReference(new PageReference('Page.CON_ContactMerge'));
+            Con_ContactMerge_CTRL ctlr = new Con_ContactMerge_CTRL(new ApexPages.StandardSetController(new list<Contact>()));
+            System.assertEquals(null, ctlr.errorMsg);
+
+            ctlr.searchText = 'test';
+            //SOSL always returns nothing in tests, unless you use the setFixedSearchResults value
+            Id[] fixedSearchResults=new Id[2];
```

Comment:
An alternative way to write these three lines is as:
`List<Id> fixedSearchResults = new List<Id>{ con.Id, con2.Id };`